Security/SessionDownloadStrategy.php

```diff
@@ -39,14 +42,31 @@ class SessionDownloadStrategy implements DownloadStrategyInterface
     protected $sessionKey = 'sonata/media/session/times';
 
     /**
-     * @param TranslatorInterface $translator
-     * @param ContainerInterface  $container
-     * @param int                 $times
+     * @var Session
      */
-    public function __construct(TranslatorInterface $translator, ContainerInterface $container, $times)
+    private $session;
+
+    /**
+     * @param TranslatorInterface        $translator
+     * @param ContainerInterface|Session $container
+     * @param int                        $times
+     */
+    public function __construct(TranslatorInterface $translator, $container, $times)
     {
+        // NEXT_MAJOR : remove this block and set session from parameter.
+        if ($container instanceof ContainerInterface) {
+            @trigger_error(
+                'Using an instance of Symfony\Component\DependencyInjection\ContainerInterface is deprecated since 
+                version 3.x and will be removed in 4.0. Use Symfony\Component\HttpFoundation\Session\Session instead.',
+                E_USER_DEPRECATED
+            );
+
+            $this->session = $container->get('session');
+        } else {
+            $this->session = $container;
```

Comment:
…and here it will be `$this->session = $session`. The new, recommended code will not look weird.